src/cpp/src/whisper_pipeline_static.cpp

```diff
@@ -288,49 +278,81 @@ std::vector<int32_t> prepare_init_ids(ov::Tensor& encoder_hidden_state,
                                 static_cast<int32_t>(config.no_timestamps_token_id)};
 }
 
-std::pair<bool, std::vector<int64_t>> full_decode(ov::Tensor& encoder_hidden_state,
-                                                  const ov::genai::WhisperGenerationConfig& config,
-                                                  ov::genai::WhisperInitializedModels& models,
-                                                  std::vector<int32_t> init_ids,
-                                                  const size_t max_new_tokens,
-                                                  const bool return_timestamps,
-                                                  ov::genai::RawPerfMetrics& raw_metrics,
-                                                  const std::shared_ptr<ov::genai::ChunkStreamerBase> streamer) {
-    int64_t output_token = decode(encoder_hidden_state, models.decoder, init_ids, config, raw_metrics, true, return_timestamps);
-    std::vector<int64_t> output_tokens{output_token};
-
-    if (!return_timestamps && streamer && streamer->put(output_token)) {
-        return {true, output_tokens};
+void stream_generated_tokens(const std::shared_ptr<ov::genai::StreamerBase> streamer_ptr,
+                             ov::genai::GenerationHandle& handle,
+                             const bool return_timestamps) {
+    if (return_timestamps || !streamer_ptr || !handle->can_read()) {
+        return;
     }
 
-    if (max_new_tokens == 1) {
-        return {false, output_tokens};
+    std::unordered_map<uint64_t, ov::genai::GenerationOutput> token = handle->read();
+    for (const auto& gen_token : token.begin()->second.generated_ids) {
+        if (streamer_ptr->put(gen_token)) {
```

Comment:
fyi `streamer->put` is deprecated. Stateful pipeline reference: https://github.com/openvinotoolkit/openvino.genai/blob/master/src/cpp/src/whisper/whisper.cpp#L70

Reply:
fixed, thanks!